ttnn/cpp/ttnn/operations/data_movement/fold/fold.hpp

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/core.hpp"
+#include "ttnn/types.hpp"
+#include "ttnn/run_operation.hpp"
+#include "ttnn/operations/core/core.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/tensor_utils.hpp"
+#include "ttnn/tensor/host_buffer/functions.hpp"
+
+#include "device/fold_device_op.hpp"
+
+namespace ttnn {
+namespace operations::data_movement {
+
+struct FoldOperation {
+    static ttnn::Tensor operator()(
+        uint8_t queue_id,
```

Comment:
we should have a version without queue_id

Reply:
Added.